hashes/src/hkdf/mod.rs

```diff
@@ -106,6 +106,29 @@ where
     }
 }
 
+impl<T: GeneralHash> fmt::Debug for Hkdf<T> {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+        use crate::{hash_newtype, sha256, sha256t, sha256t_tag};
+
+        sha256t_tag! {
+            pub struct Tag = hash_str("bitcoin_hashes1DEBUG");
+        }
+        hash_newtype! {
+            pub struct Hash(sha256::Hash);
+        }
+        let hash = sha256t::Hash::<Tag>::hash(self.prk.as_ref());
+
+        #[cfg(feature = "hex")]
+        {
+            f.debug_tuple(stringify!(Hkdf)).field(&format_args!("#{:.16}", hash)).finish()
```

Comment:
Why not just write `"Hkdf"` literal?

Also it seems to me this has different representation with and without `hex` feature.

Reply:
Yeah, in this case I'm not sure what the `hex` feature is gaining us.

Reply:
I simplified the whole thing and just used `debug_hex`. What we loose by doing without the `hex` feature is formatting options passed in by the caller and also the `Hkdf()` wrapper around the hex.

Reply:
I raised #4070

Reply:
Ugh, I've made it worse ...

Reply:
Ha that was a fun little exercise, felt like one of those programming challenge websites.